### PR TITLE
Handle known KaTeX exceptions

### LINKED ISSUE #14: Handle any kind of error, not just invalid commands
Looks like KaTeX only respects `throwOnError: false` for invalid commands (https://github.com/Khan/KaTeX/issues/475), not for any kind of error (unmatched brackets, for example, still throws an error), it would be good to fix it before releasing the next major.

Using @nikoladev's initial approach for error handling (https://github.com/talyssonoc/react-katex/pull/10/commits/285749cb6b5dd9a9ca70481ebe60b91d3623fa84#diff-04c6e90faac2675aa89e2176d2eec7d8R100) seems like a reasonable place to start with the solution.

### PR DESCRIPTION
**Pull request by dudko**
Closes #14.

### CODE REVIEW

**Comment by talyssonoc** [comment]
Could you check if it's possible to increase the coverage?

File: src/createMathComponent.js [hunk 1]
```diff
@@ -25,30 +25,39 @@ const createMathComponent = (Component, { displayMode }) => {
         const html = this.generateHtml(props);
 
         return { html, error: undefined };
-      } catch(error) {
-        return { error, html: undefined };
+      } catch (error) {
+        if (error.__proto__ === KaTeX.ParseError.prototype) {
```
Comment (talyssonoc) [inline src/createMathComponent.js]: Looks like we do the same thing for `ParseError` and `TypeError`, what do you think about we doing something like this:

```js
if(error instanceof Katex.ParseError || error instanceof TypeError) {
  return { error, html: `${error.message}` };
}
```

File: src/createMathComponent.js [hunk 2]
```diff
@@ -25,30 +25,39 @@ const createMathComponent = (Component, { displayMode }) => {
...
+        displayMode,
+        errorColor,
+        throwOnError: !!renderError
+      });
     }
 
     render() {
-      if(this.state.html) {
-        return <Component html={this.state.html} />;
+      if (this.props.renderError && this.state.error) {
```
Comment (talyssonoc) [inline src/createMathComponent.js]: We could have a `this.shouldRenderError()` and move this conditional logic to there.

**Comment by dudko** [comment]
@talyssonoc , thanks for your feedback. I worked on requested changes. If you are satisfied with them, please release a new version so `react-katex` can be finally used with latest react. :)

File: test/sharedExamples.js [hunk 3]
```diff
@@ -114,11 +114,25 @@ export default (Component, { wrapperTag, displayMode }) => {
...
+          expect(math.html()).to.equal(
+            `<${wrapperTag}>KaTeX can only parse string typed expression</${wrapperTag}>`
+          );
+        });
+      });
+
+      context('when error is caused while parsing math expression', () => {
+        it('renders error message', () => {
+          const math = shallow(
+            <Component displayMode math={incompleteFormula} />
```
Comment (talyssonoc) [inline test/sharedExamples.js]: Do we need that `displayMode` here? Besides that everything looks ok to me!

**Comment by talyssonoc** [comment]
Right now we're using `yarn.lock` instead of `package-lock.json` in the project, do you mind removing package-lock.json and updating yarn.lock instead?

File: test/sharedExamples.js (hunk 3 above)
Comment (dudko) [reply test/sharedExamples.js]: No, it's redundant. Removed.

**Pull request merged**